core/drivers/crypto/hisilicon/hpre_rsa.h

```diff
@@ -0,0 +1,47 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2022, HiSilicon Technologies Co., Ltd.
+ */
+
+#ifndef __TEE_HPRE_RSA_H__
+#define __TEE_HPRE_RSA_H__
+
+#include <stdint.h>
+#include <tee_api_types.h>
+#include <types_ext.h>
+
+#define PKCS_V1_5_MSG_MIN_LEN 11
+#define PKCS_V1_5_PS_MIN_LEN 8
+#define PKCS_V1_5_PS_POS 2
+#define PKCS_V1_5_FIXED_LEN 3
+#define PKCS_V1_5_PS_DATA 0x5a
+#define OAEP_MAX_HASH_LEN 64
+#define OAEP_MAX_DB_LEN 512
+#define PRIME_QUALITY_FLAG 1024
+#define HPRE_RSA_GEN_TOTAL_BUF_SIZE(kbytes) ((kbytes) * 7)
+#define HPRE_RSA_CRT_TOTAL_BUF_SIZE(kbytes) ((kbytes) * 6)
+#define HPRE_RSA_CRT_KEY_BUF_SIZE(kbytes) ((kbytes) >> 10)
+#define HPRE_RSA_NCRT_TOTAL_BUF_SIZE(kbytes) ((kbytes) * 4)
+
+enum pkcs_v1_5_pad_type {
+	SIGN_PAD = 1,
+	ENCRYPT_PAD = 2
+};
+
+struct hpre_rsa_msg {
+	uint8_t *pubkey;
+	paddr_t pubkey_dma;
+	uint8_t *prikey;
+	paddr_t prikey_dma;
+	uint8_t *in;
```

Comment:
It would be nice with a summary of all the expected sizes of these buffers and how they relate.

Reply:
It is different in encrypt and decrypt，also in crt and ncrt. Expected size  of these buffers and how they relate can be seen in alloc function of every mode, such as hpre_rsa_crt_decrypt_alloc.

Reply:
It sounds like it's even more important since it makes it harder to work out backward.

Reply:
Would it make sense to describe the buffer layout for `HPRE_RSA_CRT_TOTAL_BUF_SIZE()` and `HPRE_RSA_NCRT_TOTAL_BUF_SIZE()` above?

Reply:
solved